test/test_common/utility.cc

```diff
@@ -187,30 +187,27 @@ std::vector<std::string> TestUtility::split(const std::string& source, const std
 }
 
 void TestUtility::renameFile(const std::string& old_name, const std::string& new_name) {
-#if !defined(WIN32)
-  const int rc = ::rename(old_name.c_str(), new_name.c_str());
-  ASSERT_EQ(0, rc);
-#else
+#if defined(WIN32)
   // use MoveFileEx, since ::rename will not overwrite an existing file. See
   // https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/rename-wrename?view=vs-2017
   const BOOL rc = ::MoveFileEx(old_name.c_str(), new_name.c_str(), MOVEFILE_REPLACE_EXISTING);
   ASSERT_NE(0, rc);
+#else
+  const int rc = ::rename(old_name.c_str(), new_name.c_str());
```

Comment:
I think all this should be inside the filesystem abstraction (include/envoy/filesystem/filesystem.h). Can you at least add a TODO for now?

Note that the Filesystem and ThreadFactory are both part of Api::Api so you can see the convergence of all this.

Reply:
We have this broken out here since we didn't want to add production code that is only called from tests. If moving this code into the filesystem abstraction is the way to go, we will do that as a part of the filesystem PR that we are currently working on.